lib/Net/Google/Drive/Simple/V3.pm

```diff
@@ -780,6 +793,351 @@ sub upload_multipart_file {
+            qw<
+                appProperties
+                contentHints
+                contentRestrictions
+                copyRequiresWriterPermission
+                createdTime
+                description
+                folderColorRgb
+                id
+                mimeType
+                modifiedTime
+                name
+                originalFilename
+                parents
+                properties
+                shortcutDetails.targetId
+                starred
+                viewedByMeTime
+                writersCanShare
+            >
+        ],
+
+        'path'                 => $self->{'api_upload_url'},
+        'method_name'          => 'create_resumable_upload',
+        'http_method'          => HTTP_METHOD_POST(),
+        'extra_headers'        => \@extra_headers,
+        'resumable'            => 1,
+        'return_http_response' => 1,
+    };
+
+    if ( defined $options->{'enforceSingleParent'} ) {
+        LOGDIE("[$info->{'method_name'}] Creating files in multiple folders is no longer supported");
+    }
+
+    # Since a file upload can take a long time, refresh the token
+    # just in case.
+    $self->{'oauth'}->token_expire();
+
+    my $response = $self->_handle_api_method( $info, $options );
+    $response->is_success()
+        or return;
+
+    my $location = $response->header('Location');
+    return $location || '';
+}
+
+###########################################
+sub upload_file_content_single {
+###########################################
+    my ( $self, $upload_uri, $file ) = @_;
+
+    defined $upload_uri && length $upload_uri
+        or LOGDIE('upload_file_content_single() missing upload_uri');
+
+    $upload_uri =~ m{^https://.*upload_id=.+}xms
+        or LOGDIE('upload_file_content_single() upload_uri seems malformed');
+
+    defined $file && length $file
+        or LOGDIE('upload_file_content_single() missing file');
+
+    -r $file
+        or LOGDIE("upload_file_content_single() received non-existent/unreadable file: $file");
+
+    my $size = -s $file;
+    $size <= SIZE_5120GB()
+        or LOGDIE("upload_file_content_single() has a limit of 5120G, '$file' is bigger");
+
+    my $info = {
+        'path'          => $upload_uri,
+        'method_name'   => 'upload_file_content_single',
+        'http_method'   => HTTP_METHOD_PUT(),
+        'body_content'  => $self->_content_sub($file),
+    };
+
+    # Since a file upload can take a long time, refresh the token
+    # just in case.
+    $self->{'oauth'}->token_expire();
+
+    return $self->_handle_api_method( $info, {} );
+}
+
+###########################################
+sub upload_file_content_multiple {
+###########################################
+    my ( $self, $upload_uri, $file, $chunk_size ) = @_;
+
+    defined $upload_uri && length $upload_uri
+        or LOGDIE('upload_file_content_multiple() missing upload_uri');
+
+    $upload_uri =~ m{^https://.*upload_id=.+}xms
```

Comment:
consider `\bupload_id=` ?
`/i`?

Reply:
I'm not sure `/i` is technically correct, but doing it anyway.